internal/controller/server_controller.go

```diff
@@ -806,58 +779,6 @@ func (r *ServerReconciler) applyBootOrder(ctx context.Context, log logr.Logger,
 	return nil
 }
 
-func (r *ServerReconciler) applyBiosSettings(ctx context.Context, log logr.Logger, server *metalv1alpha1.Server) error {
-	serverBase := server.DeepCopy()
-	if server.Spec.BMCRef == nil && server.Spec.BMC == nil {
-		log.V(1).Info("Server has no BMC connection configured")
-		return nil
-	}
-	bmcClient, err := GetBMCClientForServer(ctx, r.Client, server, r.Insecure)
-	if err != nil {
-		return fmt.Errorf("failed to create BMC client: %w", err)
-	}
-	defer bmcClient.Logout()
-
-	version, err := bmcClient.GetBiosVersion(server.Spec.UUID)
-	if err != nil {
-		return fmt.Errorf("failed to create BMC client: %w", err)
-	}
-
-	versionMatch := false
-	diff := map[string]string{}
-	for _, bios := range server.Spec.BIOS {
-		if bios.Version == version {
-			versionMatch = true
-			for key, value := range bios.Settings {
-				if res, ok := server.Status.BIOS.Settings[key]; !ok {
-					if !ok || res != value {
-						diff[key] = value
-					}
-				}
-			}
-			reset, err := bmcClient.SetBiosAttributes(server.Spec.UUID, diff)
```

Comment:
How do we apply bios settings/ boot order now?

Reply:
Boot order: there is a question to discuss in the PR's description, whether we need to keep boot order in `Server` CR and keep its processing in server-controller, or we'll move it to `ServerBIOS`. Semantically we should move it to `ServerBIOS`.

Settings: lines 231-245 in serverbios-controller:
```go
func (r *ServerBIOSReconciler) reconcileSettingsUpdate(
	ctx context.Context,
	log logr.Logger,
	serverBIOS *metalv1alpha1.ServerBIOS,
) (ctrl.Result, error) {
	log.V(1).Info("invoking settings update job")
	jobReference, err := r.createJob(ctx, log, serverBIOS, metalv1alpha1.ApplyBIOSSettingsJobType)
	if err != nil {
		return ctrl.Result{}, err
	}
	return r.patchJobReference(ctx, log, serverBIOS, metalv1alpha1.RunningJobRef{
		Type:   metalv1alpha1.ApplyBIOSSettingsJobType,
		JobRef: jobReference,
	})
}
```

The job implementation is in progress, hopefully will finish it today and push to PR.